prow/tekton.md

```diff
@@ -0,0 +1,147 @@
+# Using Prow with Tekton Pipeline
+
+The Prow agent type `tekton-pipeline` will allow you to trigger
+execution of
+[Tekton Pipelines](https://github.com/tektoncd/pipeline) 
+with Prow!
+
+1. [Setting up Pipelines with Prow](#setting-up-pipelines-with-prow)
+2. [Configuring a `tekton-pipeline` job](#configuring-a-tekton-pipeline-job)
+
+**Warning**
```

Comment:
The other caveat is that git pipelineresources are currently not garbage collected. (I think we can fix this by updating the pipelineresources with owner references after the pipelinerun is created.)

Reply:
I implemented this initially in the controller and it worked well, but I was asked to remove it due to the multi-cluster build support. Referencing the `ProwJob` is problematic when the builds are executed in a different cluster but I don't see an issue with a reference to the `PipelineRun`.

Reply:
Added a warning about this! Btw https://github.com/tektoncd/pipeline/issues/544 is about this and it looks like this will be addressed in https://github.com/tektoncd/pipeline/issues/872 (which was your request originally @cjwagner!) - you'd have to update Prow to be able to embed PipelineResources tho :)